src/main/java/org/sagebionetworks/web/client/widget/entity/editor/ExternalImageConfigViewImpl.java

```diff
@@ -0,0 +1,93 @@
+package org.sagebionetworks.web.client.widget.entity.editor;
+
+import org.gwtbootstrap3.client.ui.TextBox;
+import org.sagebionetworks.web.client.DisplayConstants;
+import org.sagebionetworks.web.client.DisplayUtils;
+import org.sagebionetworks.web.client.SynapseJSNIUtils;
+import org.sagebionetworks.web.client.ValidationUtils;
+import org.sagebionetworks.web.client.cache.ClientCache;
+
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class ExternalImageConfigViewImpl implements ExternalImageConfigView {
+	public interface ExternalImageConfigViewImplUiBinder extends UiBinder<Widget, ExternalImageConfigViewImpl> {}
+	private Widget widget;
+	private Presenter presenter;
+	ClientCache clientCache;
+	SynapseJSNIUtils synapseJSNIUtils;
+
+	@UiField
+	TextBox nameField;
+	@UiField
+	TextBox urlField;
+
+	@Inject
+	public ExternalImageConfigViewImpl(
+			ExternalImageConfigViewImplUiBinder binder,
+			ClientCache clientCache, SynapseJSNIUtils synapseJSNIUtils
+			) {
+		widget = binder.createAndBindUi(this);
+		this.clientCache = clientCache;
+		this.synapseJSNIUtils = synapseJSNIUtils;
+	}
+
+	@Override
+	public void initView() {
+		urlField.setValue("");
+		nameField.setValue("");
+	}
+
+	@Override
+	public void checkParams() throws IllegalArgumentException {
+		if (!ValidationUtils.isValidUrl(urlField.getValue(), false))
```

Comment:
I realize this is a copy, but feel it my obligation to point out that validation should not occur in the view impl (should be in the presenter, and tested).